numpy/core/src/multiarray/mapping.c

```diff
@@ -2369,6 +2369,19 @@ mapiter_fill_info(PyArrayMapIterObject *mit, npy_index_info *indices,
             }
         }
 
+        /* Before contunuing, ensure that there are not too fancy indices */
+        if (indices[i].type & HAS_FANCY) {
+            assert(indices[i].type == HAS_FANCY ||
```

Comment:
Is this needed?

Reply:
Hmm, no, you are right. This assert is obviously always true, I confused myself and thought it might not be obvious, so I added it...